src/components/Dropdowns/StyledSelect.tsx

```diff
@@ -3,14 +3,42 @@ import React, { ReactElement, useMemo } from "react";
 import Select, { components, DropdownIndicatorProps, StylesConfig } from "react-select";
 import { StateManagerProps } from "react-select/dist/declarations/src/useStateManager";
 import styled, { css } from "styled-components";
+import { Color } from "three";
 
 import { DropdownSVG } from "../../assets";
 
 import { AppTheme, AppThemeContext } from "../AppStyle";
 
+type OptionWithColor = unknown & { color?: Color };
```

Comment:
two questions here:
1. why would color not be defined? is it only determined in the data? 
2. Isn't this more `OptionConfig`? it's not the element itself right? is there no type you can use here to extend instead of `unknown`?

Reply:
1. Color is optional, yes.
2. `Option` is a generic type that by default `react-select` sets to `unknown`. It's the underlying data that's passed into the `Select` element. I agree that the typing is confusing, so I ended up making a separate `types.ts` file and enforcing a unified `SelectItem` type instead of allowing `StyledSelect` to be totally generic.